src/sentry/models/organizationmember.py

```diff
@@ -62,10 +63,14 @@ def get_contactable_members_for_org(self, organization_id: int) -> QuerySet:
 
     def delete_expired(self, threshold: int) -> None:
         """Delete un-accepted member invitations that expired `threshold` days ago."""
-        self.filter(
-            token_expires_at__lt=threshold,
-            user_id__exact=None,
-        ).exclude(email__exact=None).delete()
+
+        orgs_with_scim = AuthProvider.objects.filter(
+            flags=F("flags").bitor(AuthProvider.flags.scim_enabled)
+        ).values_list("organization_id", flat=True)
+
+        self.filter(token_expires_at__lt=threshold, user_id__exact=None,).exclude(
+            email__exact=None
+        ).exclude(organization_id__in=orgs_with_scim).delete()
```

Comment:
Nit: You can include the email and organization_id filters in the same `.exclude` call.

Reply:
weird, i had it in the same exclude call and it deleted everything

Reply:
Ahh, you're right. Let's leave it as is